curriculum/end2end/4.js

```diff
@@ -0,0 +1,35 @@
+const path = require('path');
+const fs=require('fs');
+
+const fromDir = (myPath="/home", i=0, result = [], pathList=[myPath]) => {
+  if (!fs.existsSync(myPath)){ 
+    //console.log("ERR: Couldn't find the directory: ", myPath);
+    return result.join("\n");
+  }
+
+  let files = fs.readdirSync(pathList[0]);
```

Comment:
this is a good implementation! Good job.

Improvements:
readdirSync is not good, because it blocks execution and nothing continues. 

For anything IO related (file read/write, internet requests, etc.), you want to call functions asynchronously.

```
// function shape....
const bft = (n, cb, c=[], result=[n]) => {
...
};

// Here's how you would want to get the data:
bft('/home/song/curriculum/curriculum', (data) => {console.log(data)});
```